README.md

```diff
@@ -58,60 +96,124 @@ This deployment is available under the _kafka-inmemory_ folder and provides foll
 
 ### Deploying to OpenShift
 
-To conveniently deploy a Kakfa in-memory to OpenShift a [Template is provided](kafka-inmemory/resources/openshift-template.yaml).
-To use the template, follow these steps: 
-1. Load the template into your OpenShift cluster: `oc create -f kafka-inmemory/resources/openshift-template.yaml`
-2. Deploy Kafka and Zookeeper: `oc new-app barnabas`
+1. With the [Openshift Origin client tools][origin] installed and having logged into your cluster, 
+   create a pod using the [provided template](kafka-inmemory/resources/openshift-template.yaml) by running
+
+        oc create -f kafka-inmemory/resources/openshift-template.yaml
+
+   in your terminal. This template provides the "zookeeper" and the "kafka" deployments and the 
+   "zookeeper-service" and "kafka-service" services. 
+2. Create a new app:
+
+        oc new-app barnabas-inmemory
+
+
+### Deploying to Kubernetes
+
+1. If you don't have a cluster running, start one (e.g. by executing `minikube start`)
+2. Create the deployments and services by running:
+
+        kubectl apply -f kafka-inmemory/resources/zookeeper.yaml && \
+        kubectl apply -f kafka-inmemory/resources/zookeeper-service.yaml && \
+        kubectl apply -f kafka-inmemory/resources/kafka.yaml && \
+        kubectl apply -f kafka-inmemory/resources/kafka-service.yaml
+
+3. You can then verify that the services started using
+
+        kubectl describe all
+
 
 ## Kafka Connect
 
-This deployment adds Kafka Connect cluster which can be used with the Kafka deployments above. It is implemented as deployment with a configurable number of workers. The default image currently contains only the Connectors which are part of Apache Kafka project - `FileStreamSinkConnector` and `FileStreamSourceConnector`. The REST interface for managing the Kafka Connect cluster is exposed internally within the Kubernetes / OpenShift cluster as service `kafka-connect` on port `8083`.
+This deployment adds a [Kafka Connect][connect] cluster which can be used with either of the Kafka deployments described above. 
+It is implemented as a deployment with a configurable number of workers. 
+The default image currently contains only the Connectors distributed with Apache Kafka Connect - 
+`FileStreamSinkConnector` and `FileStreamSourceConnector`. 
+The REST interface for managing the Kafka Connect cluster is exposed internally within the Kubernetes / OpenShift 
+cluster as service `kafka-connect` on port `8083`.
 
-### Deploying to OpenShift
 
-To conveniently deploy Kafka Connect to OpenShift a [Template is provided](kafka-connect/resources/openshift-template.yaml), this could be used via `oc create -f kafka-connect/resources/openshift-template.yaml` so it is present within your project. To create a whole Kafka Connect cluster use `oc new-app barnabas-connect` *(make sure you have already deployed a Kafka broker)*.
+### Deploying to OpenShift Origin
```

Comment:
Why do you sometimes use *Deploying to OpenShift Origin* and sometimes *Deploying to OpenShift*? Maybe we should use *Deploying to OpenShift* everywhere.

Reply:
We're using (and instructing to install) the OpenShift Origin client tools.